In [4]:
import pandas
from sklearn import linear_model

c:\Users\HP\anaconda\lib\site-packages\scipy\__init__.py:138: UserWarning: A NumPy version >=1.16.5 and <1.23.0 is required for this version of SciPy (detected version 1.24.3)
  warnings.warn(f"A NumPy version >={np_minversion} and <{np_maxversion} is required for this version of "


In [13]:
df=pandas.read_csv("data.csv") # Veri setini oku. df değişkenine ata.
df.head() # İlk 5 satırı göster.

Car       Model  Volume  Weight  CO2
0      Toyoty        Aygo    1000     790   99
1  Mitsubishi  Space Star    1200    1160   95
2       Skoda      Citigo    1000     929   95
3        Fiat         500     900     865   90
4        Mini      Cooper    1500    1140  105

In [6]:
X=df[["Weight","Volume"]]
y=df["CO2"]
#Bağımsız değerler listesini büyük X harfiyle ve bağımlı değerler listesini küçük y harfiyle adlandırmak yaygındır.

In [7]:
regr = linear_model.LinearRegression() 
regr.fit(X,y) # fit fonksiyonu ile modeli eğit.

LinearRegression()

In [8]:
tahminCO2 = regr.predict([[2300, 1300]]) # 2300 kg ağırlığında ve 1300 cm3 hacminde bir arabanın CO2 emisyonunu tahmin et.

In [9]:
print(tahminCO2)

[107.2087328]


### KATSAYI (COEFFICIENT) HESAPLAMA
Katsayı, bilinmeyen bir değişkenle ilişkiyi açıklayan bir faktördür.

Örnek: if xbir değişkendir, o zaman iki 2xkatıdır x. xbilinmeyen değişkendir ve sayı 2katsayıdır.

Bu durumda ağırlığın CO2'ye karşı katsayı değerini ve CO2'ye göre hacmin katsayı değerini sorabiliriz. Aldığımız cevap(lar) bize bağımsız değerlerden birini artırır veya azaltırsak ne olacağını söyler.

In [14]:
print(regr.coef_) # Katsayıları göster.

[0.00755095 0.00780526]


Sonuç dizisi, ağırlık ve hacmin katsayı değerlerini temsil eder.

Ağırlık: 0,00755095
Hacim: 0,00780526

Bu değerler bize ağırlık 1kg artarsa ​​CO2 emisyonunun 0,00755095g arttığını söylüyor.

Ve motor hacmi (Hacim) 1 cm3 artarsa , CO2 emisyonu 0,00780526 g artar.

In [15]:
tahmin2_CO2 = regr.predict([[3300, 1300]])

print(tahmin2_CO2)

[114.75968007]


1.3 litrelik motora ve 3300 kg ağırlığa sahip bir otomobilin gittiği her kilometre için yaklaşık 115 gram CO2 salacağını tahmin etmiştik.

Bu, 0.00755095 katsayısının doğru olduğunu gösterir:

107.2087328 + (1000 * 0.00755095) = 114.75968